src/pages/workspaces/workspace/workflows/DataStepContent.js

```diff
@@ -41,10 +41,11 @@ export default class DataStepContent extends Component {
   isValidSelectionModel() {
     const { entitySelectionModel } = this.state
     const { newSetName, selectedEntities, type } = entitySelectionModel
-    const { entityType, name } = selectedEntities
+    const selectionSize = _.size(selectedEntities)
+
     return (type === EntitySelectionType.processAll ||
-      (type === EntitySelectionType.processFromSet && !!entityType && !!name) ||
-      (_.size(selectedEntities) > 0 && !!newSetName))
+      (selectionSize > 0 && !!newSetName && (type !== EntitySelectionType.chooseSet || selectionSize <= 10))
```

Comment:
You faithfully and correctly simplified this conditional. I now realize that the conditional as I originally wrote it was a little sloppy with requiring `newSetName` even if only a single row is selected. That tripped me up when trying to read and understand the intent of this change.

I decided to see what it would look like to explicitly write this longhand without de-duping any of the conditionals across selection modes.

```suggestion
      (type === EntitySelectionType.chooseRows && (selectionSize === 1 || (selectionSize > 1 && !!newSetName))) ||
      (type === EntitySelectionType.chooseSet && (selectionSize === 1 || (selectionSize > 1 && selectionSize <= 10 && !!newSetName)))
```

What do you think?